packages/fxa-auth-server/fxa-oauth-server/lib/grant.js

```diff
@@ -189,17 +190,20 @@ function generateIdToken(grant, access) {
   return jwt.sign(claims);
 }
 
-exports.generateAccessToken = async function generateAccessToken(grant) {
+exports.generateAccessToken = async function generateAccessToken(
+  grant,
+  ppidSeed
+) {
```

Comment:
Hmm...how will this interact with the desire from the webextension to fetch profile info for local display?

IIUC the webextension needs to be able to:

* Use `/token` to generate an access token with `profile` scope and the raw `uid`, which it can pass to the profile server.
* Use `/token` to generate a different access token with `secure-proxy` scope and the PPID, which it can pass to the third-party service.

But the current code will always generate tokens with the PPID, which won't be properly understood by profile-server.

I wonder if we actually do need to pass in the intended Resource Server here, both so that we can put a more specific value in the `aud` field and so that we know how to calculate the appropriate `sub`.

Alternately, you could enable PPIDs based on *scope* rather than client_id, so that anyone requesting a token with `secure-proxy` scope gets the PPID.

Reply:
Ah, interesting! There's a lot hinging on this "`jti` maps to the raw token data in the db" thing...

Reply:
Yeah, there is. It seemed like the quickest way to get JWT access tokens into prod while ensuring we don't regress on any current system behaviors.